oof ouch, my Bones (Hurtful Disposals)

Andrymeda opened the pull request:
## About The Pull Request
Makes it so that disposal pipes hurt. Because moving fast through them sounds really painful. And now, also loud. It's however, not deadly. At least not too much!

Short paths will deal from no damage to 5.
Medium ones will deal around 15.
Longer ones will deal around 25.

## Why It's Good For The Game
Disposals is currently one of the cheapest ways to escape a room. Go down the pipes silently, and you're in cargo. If you have access, tools, or just the trust of someone there, be it you a baddie or not, you're out.

I do not plan on making this damage lethal by itself, although it will depend of which area you were disposed at. The longer the path is, the more rolls it'll take to damage you, and such. But at most, it still won't be lethal.

With that in mind, it could be interesting to bring up decision making if you're an actual baddie planning on breaking into a spot and leaving. 
   With tools? You're gonna be loud and have chances to be spotted.
   With teleports? It's quick, but also loud. And a tiny more expensive, depending on which one you picked.
   Asking someone? If you're not supposed to be there in the first place, you're gonna probably get questioned, unless you got the trust of someone, be it an ally or not.
   Disposals? With this, you're gonna trade some damage for it. It could also requires you to have cargo access, but you could also just go into the maintenance area and leave through there.
It also just sounds funny from the outside, but that's a bonus.
## Changelog
:cl:
add: Disposal pipes are now hurtful. Oof, ouch, my bones.
/:cl:

Andrymeda committed:
augh ough

fuck it hurts

MMMiracles commented:
doesn't this make infinite disposal loops actually lethal

DaydreamIQ commented:
Yeah with disposal traps being a thing I don't think i'm a fan of not only being stuck in an infinite pipe but also being forced to take damage during the entire thing.

LemonInTheDark commented on code/modules/recycling/disposal/holder.dm:
```diff
@@ -87,6 +87,13 @@
 /obj/structure/disposalholder/proc/try_expel(datum/move_loop/source, succeed, visual_delay)
 	SIGNAL_HANDLER
 	if(current_pipe || !active)
+		if(!(locate(/obj/structure/disposaloutlet) in get_turf(src)) && prob(15))
+			for(var/mob/living/H in(src))
+				H.apply_damage(1, BRUTE)
+				playsound(src.loc, 'sound/effects/wounds/crack2.ogg', 50, TRUE, FALSE)
+		if(!(locate(/obj/structure/disposaloutlet) in get_turf(src)) && prob(20))
+			for(var/mob/living/H in(src))
+				playsound(src.loc, 'sound/effects/clang.ogg', 30, TRUE, FALSE)
```
`in` operators are expensive. don't duplicate them.
also it's `in src` not `in(src)`, `in` is a special keyword not a proc.

LemonInTheDark replied:
well in a perfect world you'd call this when we move to a new pipe that isn't ejecting us

LemonInTheDark replied:
alright, will bully when you're done

LemonInTheDark reviewed (changes requested):
You need to describe why this is actually good. 
Why do we want to punish people for escaping via disposals? I don't have a reason myself, so not convincing me of it just makes me want to close the pr.

LemonInTheDark commented on code/modules/recycling/disposal/holder.dm:
```diff
@@ -87,6 +87,13 @@
 /obj/structure/disposalholder/proc/try_expel(datum/move_loop/source, succeed, visual_delay)
 	SIGNAL_HANDLER
 	if(current_pipe || !active)
+		if(!(locate(/obj/structure/disposaloutlet) in get_turf(src)) && prob(15))
+			for(var/mob/living/H in(src))
+				H.apply_damage(1, BRUTE)
+				playsound(src.loc, 'sound/effects/wounds/crack2.ogg', 50, TRUE, FALSE)
+		if(!(locate(/obj/structure/disposaloutlet) in get_turf(src)) && prob(20))
+			for(var/mob/living/H in(src))
+				playsound(src.loc, 'sound/effects/clang.ogg', 30, TRUE, FALSE)
```
Oh also you shouldn't use single letter variable names. Give it a name that describes what it is in the context of this proc.
What are we doing to H, who is H, etc. It's a chance to convey extra information about the code, don't let it drop.

LemonInTheDark replied:
You are changing balance is the thing. Making disposal rides painful changes how people use them. Need to consider that and if/why you want it when you make a change is the thing. Intention doesn't affect intent.

Serxule commented:
i dont really like this because like "oh shitter in department time to beat them a little and then disposals them" but then they die from disposals and you're held accountable

Farquaar commented:
Nobody will want this, but it would be hilarious if a disposal traveller said "ouch" "oof" out loud so people standing above would hear them.

Andrymeda committed:
improved code, lowers chance by 5, damage to 2

the crack sound is so violent i think making it happen less often will make it less scary than it actually is but still funny

Andrymeda committed:
nevermind, upping the damage was a bad idea.

cahnging it back to 1.

ArcaneMusic commented on code/modules/recycling/disposal/holder.dm:
```diff
@@ -87,6 +87,13 @@
+				if(prob(10))
+					trashed_individual.apply_damage(1, BRUTE)
+					playsound(loc, 'sound/effects/wounds/crack2.ogg', 50, TRUE, FALSE)
+				if(prob(20))
```
Might want to make this an `if else`, still keep your mutually exclusive 30% of a sound effect, but introducing the chance of hearing both with the number of chances players will have to hear it would be grating most likely, especially since they can still choose to manually clang the walls by moving.

jlsnow301 commented:
Being disposaled already has its own way of punishing players- they're stunlocked for a minute+ then thrown into a room that must be broken out of. That's IF someone hasn't rigged it to kill them. 

Though this adds realism of sorts, it's not fun

castawaynont commented:
good it's funny

Striders13 commented:
disposal escape is one of the few remaining escape options that doesn't require you to powergame a full toolbelt/teleports. Lame nerf.

MrDoomBringer commented:
In a chase, skilled players are capable of using disposals to escape, and skilled players are capable of preventing someone from escaping via disposals. I don't see any reason to change that balance we have right now.

MrDoomBringer commented:
Second point - from a design perspective, we should generally promote people's ability to escape. Antags on the run from sec live longer (prolonging story conflict is good), crew on the run from antags live longer (keeping players in the game is good). Everyone wins!

Zergspower commented:
TIL we made this lethal downstream lol

Andrymeda committed:
makes damage an else-if, lowers damage chance by 5

oof ouch

Fikou commented on code/modules/recycling/disposal/holder.dm:
```diff
@@ -87,6 +87,13 @@
 /obj/structure/disposalholder/proc/try_expel(datum/move_loop/source, succeed, visual_delay)
```
this is the disposal pipe seeing if it can expel the player, damage does not fit into that

Fikou commented:
boops-the-snoot's deranged posts about 45 damage and wounding on the manuelcord made me think more about this pr so here's my thoughts:

using the environment and your knowledge of it in combat is something super cool about ss13 (thats why stuff like taking an extinguisher out of a wall cabinet and using it is awesome), dont like how this takes one of the neatest ways to use it and punishes you for it.
going in disposals you already need to stand still for a while and correctly time the use while youre being chased by someone (and they usually have a slightly faster way to cargo through the halls than you do through the disposal loop), which makes it very exciting.
imagine an antag barely getting away from sec, being at 10 hp or so and then they just get crit by random disposals damage and half a minute later come out in crit, feels extremely anticlimactic.
in terms of getting away from a situation when youre not being chased, usually this can just be achieved by a law 2 on radio, disposal wastes a bit of your time in exchange for letting you escape without needing to interact and without needing you to prepare a whole toolbelt so you can hack out of anything.
also disposaling people is funny i dont want to escalate against them i want to push them down disposals to waste a bit of a nerd's time
also we have multiple things designed to move people using disposals, see the disposal connector modsuit module or the meta cargo sorting chutes, or stuff like player built rage cage entrances, i dont like that those would damage you now

Jacquerel commented:
I think the problem you are going to run into is that there isn't any game system for having something which is both "funny damage" and also "not lethal", if you are dealing damage there's always a good chance it's going to kill someone and there's basically no situation where you can be killed by disposals and it doesn't feel sort of deflating not just for you but for anyone else involved in the interaction.

I mean I guess you could add some snowflake check which literally stops damaging them if they are close to the crit threshold, but the crit threshold is a malleable number based on your mood so that is going to be pretty ugly and unintuitive.
It could do stamina damage, but being stamcrit when you exit disposals isn't really any more exciting than just applying `paralyze` to them automatically and also doesn't really do anything. Generally speaking if you want something to be a joke it just can't deal actual damage.

flowercuco commented:
> Rather than damage you might be better off just applying a negative moodlet, or maybe a "bruised" status effect which slows your movement with duration scaling on how far you travelled.

you should get a memory of any decently long disposals trip, try to get a high score!!!

Fikou commented:
> Also mentioned the thought of capping the damage, although i still am not too sure on how to look into that

i think arbitrary caps are lame and baby stuff, if it has to damage it should always damage, its not like theres a disposal safety system that saves crit people

> relying on AI for everything is lame, in my opinion. if you're not ready to get out of the situation yourself, then, eh..?

yeah, so you have the ability to not rely on the ai by going through disposals. needing people to be prepared for every single thing that may happen in the game leads to powergame

> I also think it's funny, and i also think this will make it funnier, because it's not supposed to be lethal, if you're just bullying someone they can just leave with 20 damage that you can quickly heal up

yeah and as i fucked them up a fair bit they have more of a right to escalate to more lethal force

> I could also look into armor negating the damage of these things, and also, the damage is literally 1, it might be negligible if the path is as short as 3 tiles or so, I'd believe? also i believe the cargo sorting chutes aren't supposed to move people, but to move deliveries in the first place

sure, youre right about the rage cage 3 tile path, about the deliveries, they arent locked behind plastic flaps or anything so id argue they are just as much for humans as deliveries really

i think you should just make it play the iron bar dropping sound (clankkknk) when you switch directions if you want it to be funny or one of jacquerels ideas if you want something neat

Andrymeda committed:
raises the volume of the clang, lowers the cracks one

average of 15 damage on meta bar, and 25 on sec. i'm settling on this personally

zxaber commented:
> my plan here isn't exactly to _punish_ players who go down disposals, for the record. i really just think this would be funny. I don't want it to be lethal at all.

Instead of damage, you should just make people sick. Especially if they've eaten food recently.

moocowswag commented:
Disposals are already a major risk to use the longer a round goes on (especially for lizards) because they wrap around the entire station and there is a pretty good chance you might pop out at ground zero of a breach.

They are also extremely easy to discreetly sabotage.

optimumtact commented:
Disposals could probably use some more interesting mechanics, but this I do not think is it.

We dont' want to punish people for simply going down disposal, it is always meant to be the action of other players (either on purpose or through sheer incompetence/misunderstanding) that should ultimately cause harm.

Mothblocks commented:
yes thats fine

Mothblocks commented:
i would want to see a video of that but i dont immediately object to that

RedBaronFlyer commented:
> Nobody will want this, but it would be hilarious if a disposal traveller said "ouch" "oof" out loud so people standing above would hear them.

I'd like that as well, but honestly a solid 50% of the time I just hear

>clang x50
>_clang_ x50
>**_clang_** x50
>CLANG x50
>_CLANG_ x50
>**_CLANG_** x50
> **_CLANG_** x50
>**_CLANG_** x50
>_CLANG_ x50
>CLANG x50
>**_clang_** x50
>_clang_ x50
>clang x50

as someone in disposals passes by because they're holding down all their movement keys
